### ISSUE TITLE
Unexpected behavior of nn.DataParallel by setting model by default to training mode

### ISSUE THREAD

**Issue by sbaio** [pytorch/pytorch#28496]
## 🐛 Bug

Unexpected behavior of nn.DataParallel by setting model by default to training mode.

## To Reproduce

```
import torch
from torch import nn

model = nn.Linear(1,1)
model.eval()
print(model.training)
model = nn.DataParallel(model)
print(model.training)
```

Output is 
```
False
True
```

One would expect the training mode not to change, since wrapping a model in DataParallel does not mean setting it in training mode.

## Environment

PyTorch version: 1.2.0.dev20190715
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.1 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0: Quadro GP100
GPU 1: Quadro GP100

Nvidia driver version: 410.79
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.16.3
[pip] numpydoc==0.8.0
[pip] torch==1.2.0.dev20190627
[pip] torch-nightly==1.2.0.dev20190715
[pip] torchfile==0.1.0
[pip] torchnet==0.0.4
[pip] torchvision==0.3.0a0+2b73a48
[conda] blas                      1.0                         mkl
[conda] faiss-gpu                 1.5.1            py36h1a5d453_1    pytorch
[conda] libblas                   3.8.0                    10_mkl    conda-forge
[conda] libcblas                  3.8.0                    10_mkl    conda-forge
[conda] liblapack                 3.8.0                    10_mkl    conda-forge
[conda] liblapacke                3.8.0                    10_mkl    conda-forge
[conda] mkl                       2019.4                      243
[conda] mkl-service               2.3.0            py36he904b0f_0
[conda] mkl_fft                   1.0.14           py36ha843d7b_0
[conda] mkl_random                1.1.0            py36hd6b4f25_0
[conda] pytorch-nightly           1.2.0.dev20190627 py3.6_cuda10.0.130_cudnn7.5.1_0    pytorch
[conda] torch-nightly             1.2.0.dev20190715           <pip>
[conda] torchfile                 0.1.0                     <pip>
[conda] torchnet                  0.0.4                     <pip>
[conda] torchvision               0.3.0a0+2b73a48           <pip>

**Comment by sbaio** [comment]
Had a discussion with @fmassa about this.

Actually we are not sure what the expected result might be, same thing could be said about CPU / CUDA.

if you do
model.cuda()
model.add(nn.Linear(..))

would you expect the linear to be on the CPU or on GPU?

Actually,

```
import torch
from torch import nn

model = nn.Linear(1,1)
model.eval()
print(model.training)
model = nn.DataParallel(model)#.cuda()
print(model.training)
print(model.module.training)
```
gives the output

```
False
True
False
```

So it's just the DataParallel module wrapper which is set to training mode True, and should not affect the expected model behavior (model.training = False).

Probably it might deserve a documentation.
